db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -344,6 +346,10 @@ else if (currentConglomerateDescriptor.getIndexDescriptor().excludeDefaults()
                     && (predList == null || !predList.canSupportIndexExcludedDefaults(tableNumber,currentConglomerateDescriptor, tableDescriptor))) {
                 return false;
             }
+            else if (pastTxnId != -2) {
```

Comment:
What is magic number -2?

Reply:
-2 indicates that time-travel is not set at all, i.e. we should go ahead and use current transaction for the scan.
-1 indicates that time-travel is beyond the oldest transaction, therefore, when doing the scan, we choose `SIConstants.OLDEST_TIME_TRAVEL_TX` as past txn id.
I use now proper constants instead of these magic numbers for better readability, thanks for hinting it out.